mount/mount_solaris.go

```diff
@@ -0,0 +1,83 @@
+package mount
+
+// On Solaris we can't invoke the mount system call directly.  First,
+// the mount system call takes more than 6 arguments, and go doesn't
+// support invoking system calls that take more than 6 arguments.  Past
+// that, the mount system call is a private interfaces.  For example,
+// the arguments and data structures passed to the kernel to create an
+// nfs mount are private and can change at any time.  The only public
+// and stable interface for creating mounts on Solaris is the mount.8
+// command, so we'll invoke that here.
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	mountCmd = "/usr/sbin/mount"
```

Comment:
the following comment at the top of the file attempts to explain this:

+// On Solaris we can't invoke the mount system call directly.  First,
 +// the mount system call takes more than 6 arguments, and go doesn't
 +// support invoking system calls that take more than 6 arguments.  Past
 +// that, the mount system call is a private interfaces.  For example,
 +// the arguments and data structures passed to the kernel to create an
 +// nfs mount are private and can change at any time.  The only public
 +// and stable interface for creating mounts on Solaris is the mount.8
 +// command, so we'll invoke that here.